src/main/java/org/spongepowered/api/scoreboard/Team.java

```diff
@@ -0,0 +1,225 @@
+     */
+    void setDisplayName(Message displayName) throws IllegalArgumentException;
+
+    /**
+     * Gets the prefix prepended to the display name of users on this team.
+     *
+     * @return The prefix for this team
+     */
+    Message getPrefix();
+
+    /**
+     * Sets the prefix prepended to the display name of users on this team.
+     *
+     * @param prefix The new prefix for this team
+     * @throws IllegalArgumentException If prefix is longer than 16
+     *     characters
+     */
+    void setPrefix(Message prefix) throws IllegalArgumentException;
+
+    /**
+     * Gets the suffix appended to the display name of users on this team.
+     *
+     * @return The team's current suffix
+     */
+    Message getSuffix();
+
+    /**
+     * Sets the suffix appended to the display name of users on this team.
+     *
+     * @param suffix The new suffix for this team.
+     * @throws IllegalArgumentException If suffix is longer than 16
+     *     characters
+     */
+    void setSuffix(Message suffix) throws IllegalArgumentException;
+
+    /**
+     * Gets whether friendly fire is enabled.
+     *
+     * @return Whether friendly fire is enabled
+     */
+    boolean allowFriendlyFire();
+
+    /**
+     * Sets whether friendly fire is enabled
+     *
+     * @param enabled Whether friendly fire is enabled
+     */
+    void setAllowFriendlyFire(boolean enabled);
+
+    /**
+     * Gets whether invisible team members are shown.
+     *
+     * @return Whether to show invisible team members
+     */
+    boolean canSeeFriendlyInvisibles();
+
+    /**
+     * Sets whether invisible team members are shown.
+     *
+     * @param enabled Whether to show invisible teammates
+     */
+    void setCanSeeFriendlyInvisibles(boolean enabled);
+
+    /**
+     * Gets the {@link Visibility} which controls to who nametags
+     * of players on this team are visible to.
+     *
+     * @return The {@link Visibility} for this team's nametags.
+     */
+    Visibility getNameTagVisibility();
+
+    /**
+     * Sets the {@link Visibility} which controls to who nametags
+     * of players on this team are visible to.
+     *
+     * @param visibility The {@link Visibility} for this team's nametags.
+     */
+    void setNameTagVisibility(Visibility visibility);
+
+    /**
+     * Gets the {@link Visibility} which controls who death messages
+     * for players on this team are visible to.
+     *
+     * @return The {@link Visibility} for this team's death messages.
+     */
+    Visibility getDeathMessageVisibility();
+
+    /**
+     * Sets the {@link Visibility} which controls who death messages
+     * of players on this team are visible to.
+     *
+     * @param visibility The {@link Visibility} for this team's death messages.
+     */
+    void setDeathMessageVisibility(Visibility visibility);
+
+    /**
+     * Gets the {@link User}s on the team.
+     *
+     * @return The {@link User}s on the team
+     */
+    Set<User> getUsers();
```

Comment:
Scoreboards (the default one, at least), are persisted, and thus may contain offline players.